homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,484 @@
+                if state_key not in item:
+                    continue
+
+                guard_condition = entity_setting["guard_condition"]
+                if guard_condition and all(
+                    item.get(guard_key) == guard_value
+                    for guard_key, guard_value in guard_condition.items()
+                ):
+                    continue
+
+                entity = entity_class(
+                    coordinator,
+                    state_key,
+                    entity_setting["name"],
+                    entity_setting["kind"],
+                    item_id,
+                    entity_setting["device_class"],
+                    entity_setting["icon"],
+                    entity_setting["unit"],
+                )
+
+                entities.append(entity)
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class OmnilogicSensor(OmniLogicEntity):
+    """Defines an Omnilogic sensor entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        device_class: str,
+        icon: str,
+        unit: str,
+        item_id: tuple,
+        state_key: str,
+    ):
+        """Initialize Entities."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            item_id=item_id,
+            icon=icon,
+        )
+
+        backyard_id = item_id[:2]
+        unit_type = coordinator.data[backyard_id].get("Unit-of-Measurement")
+
+        self._state = None
+        self._unit_type = unit_type
+        self._device_class = device_class
+        self._unit = unit
+        self._state_key = state_key
+
+    @property
+    def device_class(self):
+        """Return the device class of the entity."""
+        return self._device_class
+
+    @property
+    def unit_of_measurement(self):
+        """Return the right unit of measure."""
+        return self._unit
+
+    @property
+    def force_update(self):
+        """Force update."""
+        return True
+
+
+class OmniLogicTemperatureSensor(OmnilogicSensor):
+    """Define an OmniLogic Temperature (Air/Water) Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        state_key: str,
+        name: str,
+        kind: str,
+        item_id: tuple,
+        device_class: str,
+        icon: str,
+        unit: str,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the temperature sensor."""
+        sensor_data = self.coordinator.data[self._item_id].get(self._state_key)
```

Comment:
Don't use `dict.get` when we know the key is in the dict.

Reply:
Addressed (throughout)